### DISCUSSION TITLE
compile failed on 19.1

### DISCUSSION THREAD

**Discussion by Idontknown** [community/community#20416] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello dear

I repo sync the lineage 19.1 ,then added android_device_xiaomi_mido , android_kernel_xiaomi_mido and proprietary_vendor_xiaomi from zeelog,
When I compile the lineage for mido 19.1, I got the error below, could you please help what is missed?

device/xiaomi/mido/camera/QCamera2/HAL/QCameraMuxer.cpp:1546:14: error: use of undeclared identifier 'CAMERA_CMD_LONGSHOT_ON'
        case CAMERA_CMD_LONGSHOT_ON:
             ^
device/xiaomi/mido/camera/QCamera2/HAL/QCameraMuxer.cpp:1562:14: error: use of undeclared identifier 'CAMERA_CMD_LONGSHOT_OFF'
        case CAMERA_CMD_LONGSHOT_OFF:

device/xiaomi/mido/camera/QCamera2/HAL/QCamera2HWI.cpp:5766:10: error: use of undeclared identifier 'CAMERA_CMD_LONGSHOT_OFF'                                                                                        case CAMERA_CMD_LONGSHOT_OFF:                                                                                                                                                                                         ^                                                                                                                                                                                                       device/xiaomi/mido/camera/QCamera2/HAL/QCamera2HWI.cpp:5784:10: error: use of undeclared identifier 'CAMERA_CMD_HISTOGRAM_ON'                                                                                        case CAMERA_CMD_HISTOGRAM_ON:                                                                                                                                                                                         ^                                                                                                                                                                                                       device/xiaomi/mido/camera/QCamera2/HAL/QCamera2HWI.cpp:5785:10: error: use of undeclared identifier 'CAMERA_CMD_HISTOGRAM_OFF'                                                                                       case CAMERA_CMD_HISTOGRAM_OFF:                                                                                                                                                                                        ^                                                                                                                                                                                                       device/xiaomi/mido/camera/QCamera2/HAL/QCamera2HWI.cpp:5786:38: error: use of undeclared identifier 'CAMERA_CMD_HISTOGRAM_ON'                                                                                            rc = setHistogram(command == CAMERA_CMD_HISTOGRAM_ON? true : false);                                                                                                                                                                          ^                                                                                                                                                                           device/xiaomi/mido/camera/QCamera2/HAL/QCamera2HWI.cpp:5799:10: error: use of undeclared identifier 'CAMERA_CMD_HISTOGRAM_SEND_DATA'                                                                                 case CAMERA_CMD_HISTOGRAM_SEND_DATA:                                                                                            

thanks

**Comment by FahdSeddik** [comment, marked as answer]
Hey @Idontknown ,
I think you might find an answer [here](https://forum.xda-developers.com/t/problem-while-building-lineageos-16-0.3957519). Hope it helps!

**Reply by Idontknown** [reply]
Thanks, refer to your clues, I disabled HAL to make it works.
